Bottom: 76ee8b3004b9a5f7ef828226dedfff648c5f547a
Top:    4de494bcc0d683d87139e748278c975c939e9188
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 04:01:44 +0000

Attempt to debug islandness after inserting hole bug


---

diff --git a/src/action.c b/src/action.c
index 25c74a7..0dbfd92 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1505,12 +1505,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
